x/feegrant/ante/fee.go

```diff
@@ -53,7 +53,7 @@ func (d DeductGrantedFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 
 	// ensure the grant is allowed, if we request a different fee payer
 	if feeGranter != nil && !feeGranter.Equals(feePayer) {
-		err := d.k.UseGrantedFees(ctx, feeGranter, feePayer, fee)
+		err := d.k.UseGrantedFees(ctx, feeGranter, feePayer, fee, tx.GetMsgs())
```

Comment:
I'm not sure, since we are building this only for granting `fees` may not be required extra params than we are passing now.

Reply:
@amaurym , @atheeshp  - let's create an issue to draft a design how custom fee allowance should work. The deliverable is just a concept in a documentation describing how this should work. Simple idea is:
* app developer creates his own structure
* It can be used either by passing an object method as an argument (so the object will capture whatever it needs in itself) or as an interface.

Reply:
just to brainstorm the ideas ;)

Reply:
#8937